Improves the README with more badges, Pypi statistics, contributor recognition, and star history!

pygarap opened the pull request:
This pull request updates the project description and documentation to better highlight selectolax's speed, technology stack, and community activity. It also improves the README with more badges, usage examples, contributor recognition, and links to statistics and star history.

Documentation and Branding Updates:
* Updated the project description in `README.md`, `pyproject.toml`, `setup.py`, and `docs/index.rst` to emphasize that selectolax is "an extremely fast HTML5 parser with CSS selectors, written in Cython, using Modest and Lexbor engines" [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L5-R29) [[2]](diffhunk://#diff-8d068e8797e88947c320f79e856c3e16a72b730124a8f9d7031e2c4680dfa534L4-R5) [[3]](diffhunk://#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711L11-R11) [[4]](diffhunk://#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7L195-R195).
* Enhanced the top of the `README.md` with additional badges for PyPI version, downloads, supported platforms, Python versions, CI status, license, release date, commit activity, and contributors.

Improved Examples and Formatting:
* Reformatted Python code examples in `README.md` for better readability and consistency, including improved spacing and output formatting [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L48-R72) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L75-R122) [[3]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L115-R147).

Community and Statistics Sections:
* Added sections in `README.md` to recognize contributors, display a contributors image, and link to PyPI/libraries.io/deps.dev statistics.
* Included a "Star History" section with a dynamic chart showing the project's star growth over time.

Other Minor Improvements:
* Improved line wrapping and formatting in the benchmark section of the `README.md` for clarity.… for enhanced clarity and consistency

rushter commented on docs/index.rst:
```diff
@@ -1,7 +1,8 @@
 ==========
 
-Selectolax is a fast HTML5 parser with CSS selectors.
+Selectolax is an extremely fast HTML5 parser with CSS selectors, written in Cython,
```
Please avoid using buzzwords. Just keep it as before — fast.

rushter commented on README.md:
```diff
@@ -2,9 +2,31 @@
-A fast HTML5 parser with CSS selectors using [Modest](https://github.com/lexborisov/Modest/) and [Lexbor](https://github.com/lexbor/lexbor) engines.
+---
+
+[![PyPI - Version](https://img.shields.io/pypi/v/selectolax?logo=pypi&label=Pypi&logoColor=fff)](https://pypi.org/project/selectolax)
```
That looks like a Christmas tree now.

Let's keep: current PyPI version, license, total downloads, and supported versions, CI status.

Displaying too much information does not benefit users.

pygarap commented:
@rushter Thank you for the review and the helpful comments.

I have addressed all of your concerns and pushed the updated changes.
Please let me know if you would like me to make any further adjustments.
